src/unity/toolkits/neural_net/mps_utils.mm

```diff
@@ -11,9 +11,126 @@
 #include <sys/stat.h>
 #include <memory>
 
+#import <Accelerate/Accelerate.h>
+
 namespace turi {
 namespace neural_net {
 
+// Converts from CHW to HWC
+void convert_chw_to_hwc(const float_array& image, float* out_first,
+                        float* out_last) {
+  assert(image.dim() >= 3);
+  assert(out_last - out_first == image.size());
+
+  if (image.dim() == 3) {
+    // Use Accelerate framework, interpreting each channel as an image plane.
+
+    const unsigned c = static_cast<unsigned>(image.shape()[0]);
+    const vImagePixelCount h = static_cast<vImagePixelCount>(image.shape()[1]);
+    const vImagePixelCount w = static_cast<vImagePixelCount>(image.shape()[2]);
+
+    std::vector<vImage_Buffer> image_planes(c);
+    std::vector<const vImage_Buffer*> image_plane_ptrs(c);
+    std::vector<void*> output_channels(c);
+    for (size_t i = 0; i < c; ++i) {
```

Comment:
Any reason why `c` is an `unsigned (int)` and not `size_t`, just like `i` here?

Reply:
Ah, this is mostly because I use `c` down below as the `channelCount` argument to [vImageConvert_PlanarToChunkyF](https://developer.apple.com/documentation/accelerate/1533074-vimageconvert_planartochunkyf?language=objc), which takes `unsigned int`. Implicit conversions from `unsigned int` to possibly larger unsigned integer types should be free; supplying a `size_t` value for an `unsigned int` argument could require a cast to squelch warnings.